infra/iam.tf

```diff
@@ -22,7 +22,8 @@ module "iam_assumable_role_with_oidc" {
   provider_url = module.iam_github_oidc_provider.url
 
   role_policy_arns = [
-    module.iam_policy.arn
+    module.iam_policy.arn,
+    aws_iam_policy.allow_deploy_eks_policy.arn

```

Comment:
No need for this to be separate policy.
One policy can handle both permissions needed for ECR and EKS.